woocommerce/woocommerce-gutenberg-products-block
# Add PHP8 Unit Testing

Fixes #7425

The purpose of this PR is to duplicate PHP Unit tests to run against PHP 7.4 and 8.0.

Unit testing runs under wp-env, and PHP runtime is configured under `.wp-env.json`. This PR uses a [matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) for splitting the job and generates a `.wp-env.override.json`, which effectively sets the [PHP version used on wp-env instances](https://github.com/WordPress/gutenberg/pull/25268/files).

I encountered a problem where previous cached wp-env data would interfere with the test output, so data is deleted between regular tests, and WC dev version test.

~`@wordpress/env` was also updated to v5~ and it now uses PHPUnit 9.x (our dev dependency binary) as it was an opportunity to get rid of that tech debt.
Some tests were required to be updated to work with that.




#### Accessibility




#### Other Checks


### Testing

#### Automated Tests

#### Testing



1. Inspect Github Jobs for this PR
2. Ensure Unit Testing now reports **passing tests** for PHP 7.4 and PHP 8.0


### WooCommerce Visibility




### Performance Impact



### Changelog

> n/a
